dotnet/src/Skills/Skills.Web/Bing/BingConnector.cs

```diff
@@ -23,14 +23,19 @@ public sealed class BingConnector : IWebSearchEngineConnector
     private readonly ILogger _logger;
     private readonly HttpClient _httpClient;
     private readonly string? _apiKey;
+    private readonly string? _uri;
+    private const string DefaultUri = "https://api.bing.microsoft.com/v7.0/search?q";
 
     /// <summary>
     /// Initializes a new instance of the <see cref="BingConnector"/> class.
     /// </summary>
     /// <param name="apiKey">The API key to authenticate the connector.</param>
+    /// <param name="uri"></param>
     /// <param name="logger">An optional logger to log connector-related information.</param>
-    public BingConnector(string apiKey, ILogger<BingConnector>? logger = null) :
-        this(apiKey, new HttpClient(NonDisposableHttpClientHandler.Instance, false), logger)
+    [SuppressMessage("Performance", "CA1054:Change the type of parameter 'uri'...",
```

Comment:
You could make the parameters Uri and all null and in the constructor fallback to `private static readonly Uri DefaultUri = new("https://api.bing.microsoft.com/v7.0/search?q");`